### ISSUE TITLE
Automate pagination on delegation

### ISSUE THREAD

**Issue by bonustrack** [snapshot-labs/snapshot#1379]
Currently on delegation we query 4 pages
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/utils/delegation.ts#L14
This will not be enough in future, we will require 5 then 6 etc, instead of having the number of pages hardcoded we should query only first page and keep query the next one if there is 1000 results

**Comment by rootial** [comment]
Hi @bonustrack , do you mean fetch delegations one by one until all delegations is fetched in the function `getDelegations `?

**Comment by ChaituVR** [comment]
@rootial Yes :) right now we are manually adding page numbers here https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/utils/delegation.ts#L18 we have to modify it such that It should fetch until we receive less than 1000 records 

Here is an example of how you can achieve that 
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/honeyswap/index.ts#L115-L126

**Comment by rootial** [comment]
@ChaituVR got it. i think i can take this task. thanks~
